File: android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/text/TextRange.java

package io.github.abc15018045126.sora.text;

import androidx.annotation.NonNull;

/**
 * A range made up of two {@link CharPosition} objects.
 *
 * @author abc15018045126
 */
public class TextRange {

    private CharPosition start;
    private CharPosition end;

    public TextRange(@NonNull CharPosition start, @NonNull CharPosition end) {
        this.start = start;
        this.end = end;
    }

    @NonNull
    public CharPosition getStart() {
        return start;
    }

    public void setStart(@NonNull CharPosition start) {
        this.start = start;
    }

    @NonNull
    public CharPosition getEnd() {
        return end;
    }

    public void setEnd(@NonNull CharPosition end) {
        this.end = end;
    }

    public int getStartIndex() {
        return start.index;
    }

    public int getEndIndex() {
        return end.index;
    }

    /**
     * Check if the given position is inside the range
     */
    public boolean isPositionInside(@NonNull CharPosition pos) {
        return pos.index >= start.index && pos.index < end.index;
    }

    @NonNull
    @Override
    public String toString() {
        return "TextRange{" +
                "start=" + start +
                ", end=" + end +
                '}';
    }
}

File: android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/text/TextRange.kt
package io.github.abc15018045126.sora.text


class TextRange(
    @JvmField var start: CharPosition,
    @JvmField var end: CharPosition
) {

    fun getStart(): CharPosition {
        return start
    }

    fun setStart(start: CharPosition) {
        this.start = start
    }

    fun getEnd(): CharPosition {
        return end
    }

    fun setEnd(end: CharPosition) {
        this.end = end
    }

    val startIndex: Int
        get() = start.index

    val endIndex: Int
        get() = end.index


    fun isPositionInside(pos: CharPosition): Boolean {
        return pos.index >= start.index && pos.index < end.index
    }

    override fun toString(): String {
        return "TextRange{" +
                "start=" + start +
                ", end=" + end +
                '}'
    }
}
